main/streams/cast.c

```diff
@@ -268,7 +282,7 @@ PHPAPI int _php_stream_cast(php_stream *stream, int castas, void **ret, int show
 		} else if (flags & PHP_STREAM_CAST_TRY_HARD) {
 			php_stream *newstream;
 
-			newstream = php_stream_fopen_tmpfile();
+			newstream = php_stream_fopen_tmpfile_ex(/* emit errors */ false);
```

Comment:
From what I see `PHP_STREAM_CAST_TRY_HARD` is used only during the actual casting so why do you suppress errors here?

Reply:
That is not how I see it. We do not control what flag is being passed in what context, and if someone wants to check that a stream is castable by trying hard before attempting to do so they are currently allowed to do as such.